rls/src/main/java/io/grpc/rls/LinkedHashLruCache.java

```diff
@@ -29,46 +29,30 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
-import java.util.concurrent.ScheduledExecutorService;
-import java.util.concurrent.ScheduledFuture;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicLong;
 import javax.annotation.CheckReturnValue;
 import javax.annotation.Nullable;
-import javax.annotation.concurrent.GuardedBy;
-import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * A LinkedHashLruCache implements least recently used caching where it supports access order lru
  * cache eviction while allowing entry level expiration time. When the cache reaches max capacity,
  * LruCache try to remove up to one already expired entries. If it doesn't find any expired entries,
- * it will remove based on access order of entry. On top of this, LruCache also proactively removes
- * expired entries based on configured time interval.
+ * it will remove based on access order of entry. To proactively clean up expired entries, call
+ * {@link #cleanupExpiredEntries()} (e.g., via a recurring timer).
  */
-@ThreadSafe
 abstract class LinkedHashLruCache<K, V> implements LruCache<K, V> {
 
-  private final Object lock;
-
-  @GuardedBy("lock")
   private final LinkedHashMap<K, SizedValue> delegate;
-  private final PeriodicCleaner periodicCleaner;
   private final Ticker ticker;
   private final EvictionListener<K, SizedValue> evictionListener;
-  private final AtomicLong estimatedSizeBytes = new AtomicLong();
+  private long estimatedSizeBytes;
   private long estimatedMaxSizeBytes;
 
   LinkedHashLruCache(
       final long estimatedMaxSizeBytes,
       @Nullable final EvictionListener<K, V> evictionListener,
-      int cleaningInterval,
-      TimeUnit cleaningIntervalUnit,
-      ScheduledExecutorService ses,
-      final Ticker ticker,
-      Object lock) {
+      final Ticker ticker) {
     checkState(estimatedMaxSizeBytes > 0, "max estimated cache size should be positive");
     this.estimatedMaxSizeBytes = estimatedMaxSizeBytes;
-    this.lock = checkNotNull(lock, "lock");
     this.evictionListener = new SizeHandlingEvictionListener(evictionListener);
```

Comment:
Having the evictionListener handle decrementing `estimatedMaxSizeBytes` seems like unnecessary complexity that should be eliminated.

Reply:
What does that have to do with this change? Or are you saying we should make an issue to track that clean up?

Reply:
The latter.